toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/Authenticator.kt

```diff
@@ -96,6 +179,73 @@ public fun DialogAuthenticator(
     )
 }
 
+/**
+ * Displays appropriate Authentication UI when an authentication challenge is issued.
+ *
+ * For example, when an [ArcGISAuthenticationChallenge] is issued and the [AuthenticatorState] has a corresponding
+ * [OAuthUserConfiguration], then a Custom Tab will be launched to complete the OAuth sign in.
+ *
+ * Server trust prompts and username/password prompts will be displayed in a dialog.
+ * All other prompts will be displayed differently based on their type:
+ * - OAuth and IAP challenges will be displayed in a browser.
+ * - Client certificate challenges will be displayed using the Android certificate picker.
+ *
+ * For alternate behavior, see the [Authenticator] component.
+ *
+ * @param authenticatorState the object that holds the state to handle authentication challenges.
+ * @param modifier the [Modifier] to be applied to this DialogAuthenticator.
+ * @param onBrowserChallengeRequested if not null, this will be called when a authentication challenge is pending
+ * and the browser should be launched. Use this if you wish to handle browser challenges from your own
+ * activity rather than using the [AuthenticationActivity], more information can be found in the
+ * [Authenticator Toolkit README](https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit/blob/main/toolkit/authentication/README.md).
+ * @see Authenticator
+ * @since 200.2.0
+ */
+@JvmName("DialogAuthenticatorWithBrowserAuthChallenge")
+@Composable
+public fun DialogAuthenticator(
+    authenticatorState: AuthenticatorState,
+    modifier: Modifier = Modifier,
+    onBrowserChallengeRequested: ((BrowserAuthChallenge) -> Unit)
```

Comment:
I think we should keep the use of "pending", i.e. I suggest:
```suggestion
    onPendingBrowserAuthenticationChallenge: ((BrowserAuthenticationChallenge) -> Unit)
```